### PR TITLE
Created autocomplete tests.

### PR DESCRIPTION
**Pull request by rralys**

### CODE REVIEW

**Comment by mf-code** [comment]
[5714] - lhQhaBex

**Comment by mf-code** [comment]
[5714] - wtjcIB1q

**Comment by mf-code** [comment]
[5714] - oeITUpoU

**Comment by mf-code** [comment]
[5714] - t4esMUrN

**Comment by mf-code** [comment]
[5714] - GldBsagz

**Comment by mf-code** [comment]
[5714] - Ic7ftlFM

**Comment by mf-code** [comment]
[5714] - wnXyyyQb

**Comment by mf-code** [comment]
[5714] - 7Qk8fXbS

**Comment by mf-code** [comment]
[5714] - 1UBdp1P0

**Comment by mf-code** [comment]
[5714] - 8GjkDsZv

**Comment by mf-code** [comment]
[5714] - aGHeBRYN

**Comment by mf-code** [comment]
[5714] - wB4mnpbB

**Comment by mf-code** [comment]
[5714] - CeGbQpGM

**Comment by mf-code** [comment]
[5714] - nUCy0dYl

**Comment by mf-code** [comment]
[5714] - openjdk8 - Allure report on Netlify: https://5efdb627f8483cd805e4c82e--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5714] - openjdk10 - Allure report on Netlify: https://5efdb6e6080b464f291f2e8a--musing-swartz-6fc060.netlify.app

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    public void focusOut() {
+        this.sendKeys(Keys.TAB);
+    }
+
+    public void input(String value) {
+        this.core().input(value);
+    }
+
+    @Override
+    public void clear() {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java:42]: Add @JDIAction annotation with a related description

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        WebList options = new WebList(By.cssSelector(optionsCss));
+        options.get(value).click();
+    }
+
+    @Override
+    public String placeholder() {
+        return this.core().getAttribute("placeholder");
+    }
+
+    public String displayValue() {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java:58]: Add @JDIAction annotation with a related description

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        focusOut();
+    }
+
+    public void select(String value) {
+        WebList options = new WebList(By.cssSelector(optionsCss));
+        options.get(value).click();
+    }
+
+    @Override
+    public String placeholder() {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java:54]: Add @JDIAction annotation with a related description

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    }
+
+    public String displayValue() {
+        String curAttr = getAutocompleteUniqueAttribute();
+        String displayValueXpath = displayValueCss + "[" + curAttr + "]";
+        WebList labels = new WebList(By.cssSelector(displayValueXpath));
+        return labels.get(1).getText();
+    }
+
+    public boolean isOptionHighlighted(String value) {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java:65]: Add @JDIAction annotation with a related description

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        return "true".equals(core().getAttribute("aria-invalid"));
+    }
+
+    @Override
+    public boolean isDisabled() {
+        return core().hasAttribute("disabled");
+    }
+
+    @Override
+    public boolean isEnabled() {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java:190]: Add @JDIAction annotation with a related description

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    public boolean isMandatory() {
+        return "true".equals(core().getAttribute("aria-required"));
+    }
+
+    public boolean isInvalidated() {
+        return "true".equals(core().getAttribute("aria-invalid"));
+    }
+
+    @Override
+    public boolean isDisabled() {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java:185]: Add @JDIAction annotation with a related description

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+            }
+        });
+        return curAttr[0];
+    }
+
+    public boolean isMandatory() {
+        return "true".equals(core().getAttribute("aria-required"));
+    }
+
+    public boolean isInvalidated() {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java:180]: Add @JDIAction annotation with a related description

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+        attributes.forEach(attr -> {
+            boolean doesContain = attr.contains(autocompleteAttrPrefix);
+            if (doesContain) {
+                curAttr[0] = attr;
+            }
+        });
+        return curAttr[0];
+    }
+
+    public boolean isMandatory() {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java:176]: Add @JDIAction annotation with a related description

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+    }
+
+    public WebList items(String inputValue) {
+        input(inputValue);
+        WebList options = new WebList(By.cssSelector(optionsCss));
+        click();
+        return options;
+    }
+
+    public List<String> values() {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java:110]: Add @JDIAction annotation with a related description

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+        items();
+        return getValues();
+    }
+
+    public List<String> values(String inputValue) {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java:115]: Add @JDIAction annotation with a related description

Comment (AkioFaris) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java:115]: Add JavaDoc

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+        WebList groups = new WebList(By.cssSelector(optionsGroupsCss));
+        HashMap<UIElement, List<WebElement>> groupsAndOptions = new HashMap<>();
+        groups.forEach(group -> {
+            WebList curOptions = new WebList(By.cssSelector(optionsGroupsAndValuesCss + ">" + optionsCss));
+            groupsAndOptions.put(group, curOptions.getWebElements());
+        });
+        return groupsAndOptions;
+    }
+
+    public List<String> groupsValues() {
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java:141]: Add @JDIAction annotation with a related description

File: jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/AutocompleteSection.java [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.*;
+
+public class AutocompleteSection {
+    @UI("#mat-input-0")
+    public AutoComplete autocompleteOverview;
+
+    @UI("#mat-slide-toggle-3-input")
+    public SlideToggle autocompleteDisableInput;
+
+    @UI("#mat-input-1")
```
Comment (AkioFaris) [inline jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/AutocompleteSection.java:13]: Add meaningful ids to elements as described [here](https://epam-my.sharepoint.com/personal/alexey_girin_epam_com/_layouts/15/Doc.aspx?sourcedoc=%7BE3A93BE7-D0FF-45FD-B3D2-804A31B2482C%7D&file=Angular_tests_guide.docx&action=default&mobileredirect=true&cid=f2b07436-ee73-4dad-9ab6-60f8b7b539fc) and use them in tests

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/AutocompleteTests.java [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+    @BeforeMethod
+    public void before() {
+        shouldBeLoggedIn();
+        angularPage.shouldBeOpened();
+    }
+
+    @Test
+    public void verifyAutoCompleteOverview() {
+        autocompleteSection.autocompleteOverview.setValue("F", "FloridaPopulation: 20.27M");
+        assertEquals(autocompleteSection.autocompleteOverview.getValue(), "Florida");
```
Comment (AkioFaris) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/AutocompleteTests.java:21]: Use [JDI asserts ](https://jdi-docs.github.io/jdi-light/?java#jdi-asserts]) only in basic tests

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/AutocompleteTests.java [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+
+import io.github.epam.*;
+import org.testng.annotations.*;
+
+import static io.github.com.StaticSite.*;
+import static io.github.com.pages.AngularPage.*;
+import static io.github.epam.site.steps.States.*;
+import static org.testng.Assert.*;
+
+public class AutocompleteTests extends TestsInit {
```
Comment (AkioFaris) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/AutocompleteTests.java:11]: add a test for placeholder

Comment (AkioFaris) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/AutocompleteTests.java:11]: The goal of this class is to test all autocomplete functionality with jdi asserts. If any JDI assert is missing, you should implement it.

**Comment by mf-code** [comment]
[5720] - QJG3Vasb

**Comment by mf-code** [comment]
[5720] - J5kWr3Rh

**Comment by mf-code** [comment]
[5720] - MedSohNt

**Comment by mf-code** [comment]
[5720] - JINpFxdG

**Comment by mf-code** [comment]
[5720] - bOBfzxLL

**Comment by mf-code** [comment]
[5720] - ukEibMUk

**Comment by mf-code** [comment]
[5720] - v8HEWL3E

**Comment by mf-code** [comment]
[5720] - bUcxEwdk

**Comment by mf-code** [comment]
[5720] - 38rMKbDx

**Comment by mf-code** [comment]
[5720] - E1lapWEg

**Comment by mf-code** [comment]
[5720] - vY2gaE92

**Comment by mf-code** [comment]
[5720] - W7ElrJQI

**Comment by mf-code** [comment]
[5720] - eIzuAHOt

**Comment by mf-code** [comment]
[5720] - 34C485cK

**Comment by mf-code** [comment]
[5720] - openjdk8 - Allure report on Netlify: https://5f0024da9dce2d059902bb9b--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5720] - openjdk10 - Allure report on Netlify: https://5f00257d9dce2d45ab02bd1e--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5723] - 90PqBRPv

**Comment by mf-code** [comment]
[5723] - 3qpojb7I

**Comment by mf-code** [comment]
[5723] - W0swTE1q

**Comment by mf-code** [comment]
[5723] - VgHSTQyN

**Comment by mf-code** [comment]
[5723] - 6kB4cDZo

**Comment by mf-code** [comment]
[5723] - 1wL5RuhR

**Comment by mf-code** [comment]
[5723] - J0u6GbjE

**Comment by mf-code** [comment]
[5723] - WEypRyYi

**Comment by mf-code** [comment]
[5723] - 6V3OXj00

**Comment by mf-code** [comment]
[5723] - 8WyMJcrQ

**Comment by mf-code** [comment]
[5723] - l68GeKRj

**Comment by mf-code** [comment]
[5723] - QwaneFwY

**Comment by mf-code** [comment]
[5723] - rysOsKEe

**Comment by mf-code** [comment]
[5723] - a2QEGsNT

**Comment by mf-code** [comment]
[5723] - openjdk8 - Allure report on Netlify: https://5f00bd5d8ed8b80b106a213b--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5723] - openjdk10 - Allure report on Netlify: https://5f00be3d8ed8b8cbec6a2315--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5724] - q4tHVNn6

**Comment by mf-code** [comment]
[5724] - MGD3kvpc

**Comment by mf-code** [comment]
[5724] - bHsRu0h1

**Comment by mf-code** [comment]
[5724] - QhxvjaZd

**Comment by mf-code** [comment]
[5724] - huZyrg84

**Comment by mf-code** [comment]
[5724] - zQ6SjpBy

**Comment by mf-code** [comment]
[5724] - 13pgw5ug

**Comment by mf-code** [comment]
[5724] - dgIDEG5G

**Comment by mf-code** [comment]
[5724] - F8LKDUrP

**Comment by mf-code** [comment]
[5724] - fbKGxhJr

**Comment by mf-code** [comment]
[5724] - 8P9CW9Nn

**Comment by mf-code** [comment]
[5724] - gpFOyaTV

**Comment by mf-code** [comment]
[5724] - Ce9QPFQO

**Comment by mf-code** [comment]
[5724] - yPCXVelH

**Comment by mf-code** [comment]
[5724] - openjdk8 - Allure report on Netlify: https://5f00e02091b6eb19b27f2751--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5724] - openjdk10 - Allure report on Netlify: https://5f00e126a90c989c172b07c7--musing-swartz-6fc060.netlify.app

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 15]
```diff
@@ -0,0 +1,194 @@
+package com.epam.jdi.angular.elements.common;
```
[the remark this answers is not in the document]
Comment (rralys) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java]: Done.

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 16]
```diff
@@ -0,0 +1,194 @@
...
+    private String optionsGroupsAndValuesCss = "mat-optgroup";
+    private String displayValueCss = "mat-label";
+    private String autocompleteAttrPrefix = "_ngcontent-";
+
+    public void setUniqueAutoCompleteAttribute(String value) {
+        autocompleteAttrPrefix = value;
+    }
+
+    public void click() {
+        this.core().click();
```
[the remark this answers is not in the document]
Comment (rralys) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java]: Done.

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java [hunk 17]
```diff
@@ -0,0 +1,194 @@
...
+
+    @Override
+    public void clear() {
+        this.sendKeys(Keys.CONTROL + "a");
+        this.sendKeys(Keys.DELETE);
+        focusOut();
+    }
+
+    public void select(String value) {
+        WebList options = new WebList(By.cssSelector(optionsCss));
```
[the remark this answers is not in the document]
Comment (rralys) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java]: Done.

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java (hunk 1 above)
Comment (rralys) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java]: Done.

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java (hunk 2 above)
Comment (rralys) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java]: Done.

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java (hunk 3 above)
Comment (rralys) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java]: Done.

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java (hunk 4 above)
Comment (rralys) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java]: Done.

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java (hunk 5 above)
Comment (rralys) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java]: Done.

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java (hunk 11 above)
Comment (rralys) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/elements/common/AutoComplete.java]: Done.

File: jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/AutocompleteSection.java (hunk 12 above)
Comment (rralys) [reply jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/AutocompleteSection.java]: Done.

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/AutocompleteTests.java (hunk 13 above)
Comment (rralys) [reply jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/AutocompleteTests.java]: Updated with JDI Asserts.

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/AutocompleteTests.java (hunk 14 above)
Comment (rralys) [reply jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/AutocompleteTests.java:12]: Added placeholder steps to tests.

Comment (rralys) [reply jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/AutocompleteTests.java:12]: Done.

**Comment by mf-code** [comment]
[5727] - Ec60JgPB

**Comment by mf-code** [comment]
[5727] - cDYNLu2n

**Comment by mf-code** [comment]
[5727] - U8d6Gjrn

**Comment by mf-code** [comment]
[5727] - qCHg2EQQ

**Comment by mf-code** [comment]
[5727] - XRmUThA1

**Comment by mf-code** [comment]
[5727] - 3muCUc2I

**Comment by mf-code** [comment]
[5727] - D1vsngOy

**Comment by mf-code** [comment]
[5727] - e9z5yfk6

**Comment by mf-code** [comment]
[5727] - BbEdEZxl

**Comment by mf-code** [comment]
[5727] - ITIGtFIC

**Comment by mf-code** [comment]
[5727] - GmBRzYkF

**Comment by mf-code** [comment]
[5727] - 7R58OLDl

**Comment by mf-code** [comment]
[5728] - v4HgqNKk

**Comment by mf-code** [comment]
[5727] - uYf2oKL7

**Comment by mf-code** [comment]
[5728] - OV557v1M

**Comment by mf-code** [comment]
[5728] - pmv8Kk0l

**Comment by mf-code** [comment]
[5727] - wCC1JLVm

**Comment by mf-code** [comment]
[5728] - E12na27e

**Comment by mf-code** [comment]
[5728] - dSPRZSGt

**Comment by mf-code** [comment]
[5728] - Xqo6brr5

**Comment by mf-code** [comment]
[5727] - openjdk8 - Allure report on Netlify: https://5f02c3ce8ed8b8e15d6a2414--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5727] - openjdk10 - Allure report on Netlify: https://5f02c4b29dce2d234802bbd2--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5728] - jIHyeOSi

**Comment by mf-code** [comment]
[5728] - qZcBTGRc

**Comment by mf-code** [comment]
[5728] - vLr2J0XZ

**Comment by mf-code** [comment]
[5728] - B7WzsGFh

**Comment by mf-code** [comment]
[5728] - g9jdKqUH

**Comment by mf-code** [comment]
[5728] - Ruo4lUVh

**Comment by mf-code** [comment]
[5729] - xUPcCU5L

**Comment by mf-code** [comment]
[5728] - xWielUOY

**Comment by mf-code** [comment]
[5729] - rFfvaScE

**Comment by mf-code** [comment]
[5729] - kDrWNOYO

**Comment by mf-code** [comment]
[5728] - LB6k9blO

**Comment by mf-code** [comment]
[5729] - MXYPDqfq

**Comment by mf-code** [comment]
[5729] - MGbDJ92C

**Comment by mf-code** [comment]
[5729] - AsM6B31a

**Comment by mf-code** [comment]
[5728] - openjdk8 - Allure report on Netlify: https://5f02c9bd93e8b550ce62780f--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5729] - 2utlP1WD

**Comment by mf-code** [comment]
[5729] - ScaEQeLY

**Comment by mf-code** [comment]
[5728] - openjdk10 - Allure report on Netlify: https://5f02ca758ed8b851726a2095--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5729] - cButK3V0

**Comment by mf-code** [comment]
[5729] - N16Avjv1

**Comment by mf-code** [comment]
[5729] - By4devNs

**Comment by mf-code** [comment]
[5729] - oonZPMg4

**Comment by mf-code** [comment]
[5729] - lUVtabHY

**Comment by mf-code** [comment]
[5729] - oK91pAyF

**Comment by mf-code** [comment]
[5729] - openjdk8 - Allure report on Netlify: https://5f02ceb772ef4b7fecf9e3af--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5729] - openjdk10 - Allure report on Netlify: https://5f02cf6b72ef4bd0a2f9e019--musing-swartz-6fc060.netlify.app

File: jdi-light-angular/src/main/java/com/epam/jdi/angular/asserts/AutoCompleteAssert.java [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+
+    @JDIAction("Assert that '{name}' has placeholder {0}")
+    public AutoCompleteAssert placeholder(String expected) {
+        jdiAssert(element.placeholder(), Matchers.is(expected));
+        return this;
+    }
+
+    @JDIAction("Assert that '{name} is mandatory")
+    public AutoCompleteAssert mandatory() {
+        jdiAssert(element.isMandatory().toString(), Matchers.is("true"));
```
Comment (AkioFaris) [inline jdi-light-angular/src/main/java/com/epam/jdi/angular/asserts/AutoCompleteAssert.java:49]: Is toString() necessary here? Can we compare Boolean values? (And in other similar places)

Comment (rralys) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/asserts/AutoCompleteAssert.java:49]: I tried to compare boolean values, however matcher requires object, not a primitive, hence conversion to string.

Comment (rralys) [reply jdi-light-angular/src/main/java/com/epam/jdi/angular/asserts/AutoCompleteAssert.java:49]: Found a way to use booleans.

File: jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/AutocompleteSection.java [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.epam.jdi.angular.elements.common.*;
+import com.epam.jdi.angular.elements.complex.*;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.*;
+
+public class AutocompleteSection {
+    @UI("#autocomplete-overview-input")
+    public AutoComplete autocompleteOverview;
+
+    @UI("#mat-slide-toggle-3-input")
```
[the remark this answers is not in the document]
Comment (rralys) [reply jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/AutocompleteSection.java]: Done.

**Comment by mf-code** [comment]
[5734] - Xt5AEzpQ

**Comment by mf-code** [comment]
[5734] - rk0AmYoh

**Comment by mf-code** [comment]
[5734] - UA0u899J

**Comment by mf-code** [comment]
[5734] - ZEY3NGZU

**Comment by mf-code** [comment]
[5734] - VxORE49p

**Comment by mf-code** [comment]
[5734] - fnDvk2yZ

**Comment by mf-code** [comment]
[5734] - AxWTf9PH

**Comment by mf-code** [comment]
[5734] - 9rxZ2QPP

**Comment by mf-code** [comment]
[5734] - XUKjXCM6

**Comment by mf-code** [comment]
[5734] - RPf9WspE

**Comment by mf-code** [comment]
[5734] - 7FzXE0X6

**Comment by mf-code** [comment]
[5734] - uLhdcVSd

**Comment by mf-code** [comment]
[5734] - psVx9xSy

**Comment by mf-code** [comment]
[5734] - vMufoaTx

**Comment by mf-code** [comment]
[5734] - openjdk8 - Allure report on Netlify: https://5f04452d46e848e60a4a2f88--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5734] - openjdk10 - Allure report on Netlify: https://5f0445e4c7522c8ee750a675--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5748] - cs0Fkq83

**Comment by mf-code** [comment]
[5748] - iNQRfaR0

**Comment by mf-code** [comment]
[5748] - 5CerYX6N

**Comment by mf-code** [comment]
[5748] - mIxW0jQy

**Comment by mf-code** [comment]
[5748] - bxGEI81I

**Comment by mf-code** [comment]
[5748] - UCdH3Mlp

**Comment by mf-code** [comment]
[5748] - 6gxAg8JJ

**Comment by mf-code** [comment]
[5748] - UMTM4tZY

**Comment by mf-code** [comment]
[5748] - BCM7XuiG

**Comment by mf-code** [comment]
[5748] - AKKqQpnP

**Comment by mf-code** [comment]
[5748] - oNHtVEvj

**Comment by mf-code** [comment]
[5748] - ZGfvJC6e

**Comment by mf-code** [comment]
[5748] - GtVTLjwg

**Comment by mf-code** [comment]
[5748] - lo18twT8

**Comment by mf-code** [comment]
[5748] - openjdk8 - Allure report on Netlify: https://5f06dca1c7235dafa5bb7bc6--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5748] - openjdk10 - Allure report on Netlify: https://5f06dd66322f0ba9462e7dc1--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5750] - ESjWrhCH

**Comment by mf-code** [comment]
[5750] - FPnqeEb1

**Comment by mf-code** [comment]
[5750] - 18UI4pof

**Comment by mf-code** [comment]
[5750] - UZNonwmT

**Comment by mf-code** [comment]
[5750] - 6dB9v6zV

**Comment by mf-code** [comment]
[5750] - 5XhqzGPN

**Comment by mf-code** [comment]
[5750] - lCEuCI3C

**Comment by mf-code** [comment]
[5750] - bTZI5nNf

**Comment by mf-code** [comment]
[5750] - p1KC1EZs

**Comment by mf-code** [comment]
[5750] - B3xbqm25

**Comment by mf-code** [comment]
[5750] - KeIcj1pT

**Comment by mf-code** [comment]
[5750] - n91C6sZa

**Comment by mf-code** [comment]
[5750] - UX3CHXFc

**Comment by mf-code** [comment]
[5750] - qUTNM4v1

**Comment by mf-code** [comment]
[5750] - openjdk8 - Allure report on Netlify: https://5f06ee7f1d0b73c2ecb226bd--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5750] - openjdk10 - Allure report on Netlify: https://5f06ef69ae2fc5903dd32553--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5752] - WR4UICtC

**Comment by mf-code** [comment]
[5752] - 3zxvr27t

**Comment by mf-code** [comment]
[5752] - rrlfA2wM

**Comment by mf-code** [comment]
[5752] - lxR6svmA

**Comment by mf-code** [comment]
[5752] - aHilFS63

**Comment by mf-code** [comment]
[5752] - ri153jPc

**Comment by mf-code** [comment]
[5752] - AVt2XQgB

**Comment by mf-code** [comment]
[5752] - 2yVwMZCO

**Comment by mf-code** [comment]
[5752] - cVxdnmII

**Comment by mf-code** [comment]
[5752] - mIxVnK8n

**Comment by mf-code** [comment]
[5752] - hE6o6y7m

**Comment by mf-code** [comment]
[5752] - ZtddhI8p

**Comment by mf-code** [comment]
[5752] - ChxCB0Bs

**Comment by mf-code** [comment]
[5752] - icjsjOyc

**Comment by mf-code** [comment]
[5752] - openjdk8 - Allure report on Netlify: https://5f06fa77ba8a4e0ecd9f6e92--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5752] - openjdk10 - Allure report on Netlify: https://5f06fb249c06597991ec602e--musing-swartz-6fc060.netlify.app

**Pull request merged**